Azure OS Disk is Always Permium SSD

## Description

Trying to create a VM with Standard SSD disk in Azure.

## Expected behavior

Just a new VM with a standard SSD disk in Azure.

## Actual behavior

`Install-Lab` is trying to create a machine with a premium SSD disk. In my subscription it is prohibited, so a VM is not properly created.

```output
14:20:53|00:00:00|00:00:00.000| Initialization
14:20:53|00:00:00|00:00:00.336| - Host operating system version: 'Microsoft Windows Server 2022 Datacenter, 10.0.20348.0'
14:20:53|00:00:00|00:00:00.343| - Creating new lab definition with name 'SpDscDevInfraCIWin20[12](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:13)R2'
14:20:53|00:00:00|00:00:00.381| - Location of lab definition files will be 'C:\ProgramData\AutomatedLab/Labs/SpDscDevInfraCIWin2012R2'
14:20:53|00:00:00|00:00:00.000| - Performing update of SysInternals suite and lab sources directory now
14:21:02|00:00:09|00:00:08.949|   - SysInternals Suite has been updated and placed in 'C:\LabSources\Tools\SysInternals'
14:21:02|00:00:09|00:00:09.023| - Location of LabSources folder is 'C:\LabSources'
14:21:03|00:00:10|00:00:00.000| - Adding Azure subscription data
14:21:04|00:00:10|00:00:00.525|   - Using Azure Subscription 'RND-DEV-SSG_SWEDEN_EVAL_MS_SHP_SE' (***)
14:21:[13](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:14)|00:00:20|00:00:09.620|   - Using Azure Location 'West Europe'
[14](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:15):21:13|00:00:20|00:00:09.626|   - Trying to locate or create default resource group
14:21:[15](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:16)|00:00:22|00:00:12.376|   - Querying available vm sizes for Azure location 'West Europe'
14:21:37|00:00:44|00:00:00.000|   - Looking for Azure LabSources inside subscription 'RND-DEV-SSG_SWEDEN_EVAL_MS_SHP_SE'
14:21:39|00:00:46|00:00:01.700|     - Azure LabSources verified / created
14:21:39|00:00:46|00:00:00.000|   - Auto-adding ISO files from Azure labsources share
14:21:46|00:00:53|00:00:06.929|     - Added 'https://automatedlabsourcesggnhc.file.core.windows.net/labsources/ISOs/SQLServer20[16](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:17)-x64-ENU.iso'
14:21:46|00:00:53|00:00:06.934|     - Added 'https://automatedlabsourcesggnhc.file.core.windows.net/labsources/ISOs/SQLServer20[19](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:20)-x64-ENU-Dev.iso'
14:21:46|00:00:53|00:00:06.941|     - Done
14:21:46|00:00:53|00:00:07.635|   - Querying available operating system images from Azure
14:23:42|00:02:49|00:02:03.5[20](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:21)|   - Azure default resource group name will be 'SpDscDevInfraCIWin2012R2'
14:23:42|00:02:49|00:02:03.525|   - Azure data center location will be 'West Europe'
14:23:42|00:02:49|00:02:03.528|   - Finished adding Azure subscription data
14:23:43|00:02:50|00:00:00.000| - Adding Azure machine definition 'swazspfe00'   
14:23:43|00:02:50|00:00:00.138|   - No virtual networks specified. Creating a network automatically
14:23:45|00:02:52|00:00:02.036|   - Creating virtual network with name 'SpDscDevInfraCIWin2012R2' and address space '192.168.3.1/24'
14:23:46|00:02:53|00:00:03.745|   - Done
14:23:47|00:02:54|00:00:04.136| Done
14:23:47|00:02:54|00:00:00.000| Validating lab definition
14:23:53|00:03:00|00:00:05.488| - Success
14:23:53|00:03:00|00:00:05.760| Lab 'SpDscDevInfraCIWin2012R2' hosted on 'Azure' imported with 1 machines
14:23:58|00:03:05|00:00:10.381| done
14:23:59|00:03:06|00:00:00.000| Creating VMs
14:24:00|00:03:07|00:00:00.000| - Waiting for all machines to finish installing
14:24:01|00:03:08|00:00:01.436|   - Deploying new resource group with template C:\ProgramData\AutomatedLab\Labs\SpDscDevInfraCIWin2012R2\armtemplate.json
11/29/20[22](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:23) 14:24:49 Exception in lab provisioning:
New-LabVM...leaving: Deployment of resource group 'AutomatedLab.Lab' failed with 'The Resource 'Microsoft.Network/publicIPAddresses/SpDscDevInfraCIWin2012R2SpDscDevInfraCIWin2012R2lbfrontendip' under resource group 'SpDscDevInfraCIWin2012R2' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
StatusCode: 404
ReasonPhrase: Not Found
ErrorCode: ResourceNotFound
ErrorMessage: The Resource 'Microsoft.Network/publicIPAddresses/SpDscDevInfraCIWin2012R2SpDscDevInfraCIWin2012R2lbfrontendip' under resource group 'SpDscDevInfraCIWin2012R2' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
OperationID : 4eef4d6b-5766-4[23](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:24)a-8197-d98f83e8bc6e 2:[24](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:25):49 PM - The deployment 'e359c3b6-a49c-43ad-a7fe-324ede401ad3' failed with error(s). Showing 1 out of 1 error(s).
Status Message: Resource 'swazspfe00_OsDisk_1_f1edab59[27](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:28)86439a8d43c8c3359966ca' was disallowed by policy. Reasons: 'INFORMATION : THE USE OF PREMIUM SSD IS DENIED. PLEASE ENSURE THAT ON DISKS TAB EITHER STANDARD HDD OR STANDARD SSD IS SELECTED OR DEPLOYMENT MAY FAIL.
PLEASE READ DEP DOCUMENTATION FOR MORE INFORMATION'. See error details for policy resource IDs.  Target: '/subscriptions/***/resourceGroups/SpDscDevInfraCIWin2012R2/providers/Microsoft.Compute/disks/swazspfe00_OsDisk_1_f1edab592786439a8d43c8c3359966ca'. (Code:RequestDisallowedByPolicy)

CorrelationId: a6c5a9c8-74c6-483c-ac91-9589[30](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:31)7c87d6 2:24:49 PM - The deployment 'e359c3b6-a49c-43ad-a7fe-[32](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:33)4ede401ad3' failed with error(s). Showing 1 out of 1 error(s).
Status Message: Resource 'swazspfe00_OsDisk_1_f1edab592786439a8d43c8c[33](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:34)59966ca' was disallowed by policy. Reasons: 'INFORMATION : THE USE OF PREMIUM SSD IS DENIED. PLEASE ENSURE THAT ON DISKS TAB EITHER STANDARD HDD OR STANDARD SSD IS SELECTED OR DEPLOYMENT MAY FAIL.
PLEASE READ DEP DOCUMENTATION FOR MORE INFORMATION'. See error details for policy resource IDs.  Target: '/subscriptions/***/resourceGroups/SpDscDevInfraCIWin2012R2/providers/Microsoft.Compute/disks/swazspfe00_OsDisk_1_f1edab592786439a8d43c8c3[35](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:36)9966ca'. (Code:RequestDisallowedByPolicy)

CorrelationId: a6c5a9c8-74c6-483c-ac91-9589307c87d6'
Exception: D:\a\SharePointDsc-Dev-Infra\SharePointDsc-Dev-Infra\src\soe\new.ps1:198
Line |
 198 |      Throw $_.Exception.Message;
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~
     | New-LabVM...leaving: Deployment of resource group 'AutomatedLab.Lab' failed with 'The Resource
     | 'Microsoft.Network/publicIPAddresses/SpDscDevInfraCIWin2012R2SpDscDevInfraCIWin2012R2lbfrontendip' under
     | resource group 'SpDscDevInfraCIWin2012R2' was not found. For more details please go to
     | https://aka.ms/ARMResourceNotFoundFix StatusCode: 404 ReasonPhrase: Not Found ErrorCode: ResourceNotFound
     | ErrorMessage: The Resource
     | 'Microsoft.Network/publicIPAddresses/SpDscDevInfraCIWin2012R2SpDscDevInfraCIWin2012R2lbfrontendip' under
     | resource group 'SpDscDevInfraCIWin2012R2' was not found. For more details please go to
     | https://aka.ms/ARMResourceNotFoundFix OperationID : 4eef4d6b-5766-423a-8197-d98f83e8bc6e 2:24:49 PM - The
     | deployment 'e359c3b6-a49c-43ad-a7fe-324ede401ad3' failed with error(s). Showing 1 out of 1 error(s). Status
     | Message: Resource 'swazspfe00_OsDisk_1_f1edab5927864[39](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:40)a8d43c8c3359966ca' was disallowed by policy. Reasons:
     | 'INFORMATION : THE USE OF PREMIUM SSD IS DENIED. PLEASE ENSURE THAT ON DISKS TAB EITHER STANDARD HDD OR STANDARD
     | SSD IS SELECTED OR DEPLOYMENT MAY FAIL. PLEASE READ DEP DOCUMENTATION FOR MORE INFORMATION'. See error details
     | for policy resource IDs.  Target:
     | '/subscriptions/***/resourceGroups/SpDscDevInfraCIWin2012R2/providers/Microsoft.Compute/disks/swazspfe00_OsDisk_1_f1edab592786439a8d43c8c3359966ca'. (Code:RequestDisallowedByPolicy)  CorrelationId: a6c5a9c8-74c6-483c-ac91-9589307c87d6 2:24:49 PM - The deployment 'e359c3b6-a49c-43ad-a7fe-324ede[40](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:41)1ad3' failed with error(s). Showing 1 out of 1 error(s). Status Message: Resource 'swazspfe00_OsDisk_1_f1edab592786[43](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:44)9a8d43c8c3359966ca' was disallowed by policy. Reasons: 'INFORMATION : THE USE OF PREMIUM SSD IS DENIED. PLEASE ENSURE THAT ON DISKS TAB EITHER STANDARD HDD OR STANDARD SSD IS SELECTED OR DEPLOYMENT MAY FAIL. PLEASE READ DEP DOCUMENTATION FOR MORE INFORMATION'. See error details for policy resource IDs.  Target: '/subscriptions/***/resourceGroups/SpDscDevInfraCIWin2012R2/providers/Microsoft.Compute/disks/swazspfe00_OsDisk_1_f1edab592786439a8d43c8c3359966ca'. (Code:RequestDisallowedByPolicy)  CorrelationId: a6c5a9c8-74c6-[48](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:49)3c-ac91-9[58](https://github.com/alexsapozhkov/SharePointDsc-Dev-Infra/actions/runs/3574901205/jobs/6010762636#step:4:59)9307c87d6'

Error: Process completed with exit code 1.
```

```json
{
  "contentVersion": "1.0.0.0",
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "resources": [
    {
      "name": "nsg",
      "tags": {
        "AutomatedLab": "SharePointDscDev123",
        "CreationTime": "2022-11-30 11:46:39"
      },
      "properties": {
        "securityRules": [
          {
            "properties": {
              "sourceAddressPrefixes": [],
              "priority": 100,
              "sourceAddressPrefix": "*",
              "destinationPortRanges": [
                "22",
                "3389",
                "5985",
                "5986"
              ],
              "direction": "Inbound",
              "destinationAddressPrefix": "VirtualNetwork",
              "sourcePortRange": "*",
              "access": "Allow",
              "destinationAddressPrefixes": [],
              "protocol": "TCP",
              "sourcePortRanges": []
            },
            "name": "NecessaryPorts"
          },
          {
            "properties": {
              "sourceAddressPrefixes": [],
              "priority": 101,
              "sourceAddressPrefix": "*",
              "destinationPortRanges": [
                "443"
              ],
              "direction": "Inbound",
              "destinationAddressPrefix": "*",
              "sourcePortRange": "*",
              "access": "Allow",
              "destinationAddressPrefixes": [],
              "protocol": "TCP",
              "sourcePortRanges": []
            },
            "name": "BastionIn"
          },
          {
            "properties": {
              "sourceAddressPrefixes": [],
              "priority": 100,
              "sourceAddressPrefix": "*",
              "destinationPortRanges": [
                "443"
              ],
              "direction": "Outbound",
              "destinationAddressPrefix": "AzureCloud",
              "sourcePortRange": "*",
              "access": "Allow",
              "destinationAddressPrefixes": [],
              "protocol": "TCP",
              "sourcePortRanges": []
            },
            "name": "BastionMgmtOut"
          },
          {
            "properties": {
              "sourceAddressPrefixes": [],
              "priority": 101,
              "sourceAddressPrefix": "*",
              "destinationPortRanges": [
                "3389",
                "22"
              ],
              "direction": "Outbound",
              "destinationAddressPrefix": "VirtualNetwork",
              "sourcePortRange": "*",
              "access": "Allow",
              "destinationAddressPrefixes": [],
              "protocol": "TCP",
              "sourcePortRanges": []
            },
            "name": "BastionRdsOut"
          }
        ]
      },
      "type": "Microsoft.Network/networkSecurityGroups",
      "apiVersion": "2022-01-01",
      "location": "[resourceGroup().location]"
    },
    {
      "name": "SharePointDscDev123",
      "tags": {
        "AutomatedLab": "SharePointDscDev123",
        "CreationTime": "2022-11-30 11:46:39"
      },
      "dependsOn": [
        "[resourceId('Microsoft.Network/networkSecurityGroups', 'nsg')]"
      ],
      "properties": {
        "subnets": [
          {
            "properties": {
              "addressPrefix": "192.168.6.0/24",
              "networkSecurityGroup": {
                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', 'nsg')]"
              }
            },
            "name": "default"
          }
        ],
        "addressSpace": {
          "addressPrefixes": [
            "192.168.6.0/24"
          ]
        },
        "dhcpOptions": {
          "dnsServers": []
        }
      },
      "type": "Microsoft.Network/virtualNetworks",
      "apiVersion": "2022-01-01",
      "location": "[resourceGroup().location]"
    },
    {
      "name": "lbip0",
      "tags": {
        "Vnet": "SharePointDscDev123",
        "AutomatedLab": "SharePointDscDev123",
        "CreationTime": "2022-11-30 11:46:39"
      },
      "sku": {
        "name": "Standard"
      },
      "properties": {
        "publicIPAllocationMethod": "static",
        "dnsSettings": {
          "domainNameLabel": "[concat('al-', uniqueString(resourceGroup().id))]"
        }
      },
      "type": "Microsoft.Network/publicIPAddresses",
      "apiVersion": "2022-01-01",
      "location": "[resourceGroup().location]"
    },
    {
      "dependsOn": [
        "[resourceId('Microsoft.Network/publicIPAddresses', 'lbip0')]"
      ],
      "properties": {
        "frontendIPConfigurations": [
          {
            "properties": {
              "publicIPAddress": {
                "id": "[resourceId('Microsoft.Network/publicIPAddresses', 'lbip0')]"
              }
            },
            "name": "0lbfc"
          }
        ],
        "backendAddressPools": [
          {
            "name": "0lbbc"
          }
        ],
        "outboundRules": [
          {
            "properties": {
              "enableTcpReset": true,
              "allocatedOutboundPorts": 0,
              "idleTimeoutInMinutes": 4,
              "backendAddressPool": {
                "id": "[concat(resourceId('Microsoft.Network/loadBalancers', 'lb0'), '/backendAddressPools/0lbbc')]"
              },
              "frontendIPConfigurations": [
                {
                  "id": "[resourceId('Microsoft.Network/loadBalancers/frontendIPConfigurations', 'lb0', '0lbfc')]"
                }
              ],
              "protocol": "All"
            },
            "name": "InternetAccess"
          }
        ],
        "inboundNatRules": [
          {
            "properties": {
              "frontendIPConfiguration": {
                "id": "[resourceId('Microsoft.Network/loadBalancers/frontendIPConfigurations', 'lb0', '0lbfc')]"
              },
              "enableFloatingIP": false,
              "backendPort": 3389,
              "protocol": "Tcp",
              "frontendPort": 5001
            },
            "name": "swazspfe00rdpin"
          },
          {
            "properties": {
              "frontendIPConfiguration": {
                "id": "[resourceId('Microsoft.Network/loadBalancers/frontendIPConfigurations', 'lb0', '0lbfc')]"
              },
              "enableFloatingIP": false,
              "backendPort": 5985,
              "protocol": "Tcp",
              "frontendPort": 5002
            },
            "name": "swazspfe00winrmin"
          },
          {
            "properties": {
              "frontendIPConfiguration": {
                "id": "[resourceId('Microsoft.Network/loadBalancers/frontendIPConfigurations', 'lb0', '0lbfc')]"
              },
              "enableFloatingIP": false,
              "backendPort": 5986,
              "protocol": "Tcp",
              "frontendPort": 5003
            },
            "name": "swazspfe00winrmhttpsin"
          },
          {
            "properties": {
              "frontendIPConfiguration": {
                "id": "[resourceId('Microsoft.Network/loadBalancers/frontendIPConfigurations', 'lb0', '0lbfc')]"
              },
              "enableFloatingIP": false,
              "backendPort": 22,
              "protocol": "Tcp",
              "frontendPort": 5004
            },
            "name": "swazspfe00sshin"
          }
        ]
      },
      "apiVersion": "2022-01-01",
      "sku": {
        "name": "Standard"
      },
      "tags": {
        "Vnet": "SharePointDscDev123",
        "AutomatedLab": "SharePointDscDev123",
        "CreationTime": "2022-11-30 11:46:39"
      },
      "location": "[resourceGroup().location]",
      "type": "Microsoft.Network/loadBalancers",
      "name": "lb0"
    },
    {
      "name": "SharePointDscDev123",
      "tags": {
        "AutomatedLab": "SharePointDscDev123",
        "CreationTime": "2022-11-30 11:46:39"
      },
      "sku": {
        "name": "Aligned"
      },
      "properties": {
        "platformFaultDomainCount": 2,
        "platformUpdateDomainCount": 2
      },
      "type": "Microsoft.Compute/availabilitySets",
      "apiVersion": "2022-03-01",
      "location": "[resourceGroup().location]"
    },
    {
      "name": "swazspfe00nic0",
      "tags": {
        "AutomatedLab": "SharePointDscDev123",
        "CreationTime": "2022-11-30 11:46:39"
      },
      "dependsOn": [
        "[resourceId('Microsoft.Network/virtualNetworks', 'SharePointDscDev123')]",
        "[resourceId('Microsoft.Network/loadBalancers', 'lb0')]"
      ],
      "properties": {
        "enableAcceleratedNetworking": false,
        "ipConfigurations": [
          {
            "name": "ipconfig1",
            "properties": {
              "loadBalancerInboundNatRules": [
                {
                  "id": "[concat(resourceId('Microsoft.Network/loadBalancers', 'lb0'),'/inboundNatRules/swazspfe00rdpin')]"
                },
                {
                  "id": "[concat(resourceId('Microsoft.Network/loadBalancers', 'lb0'),'/inboundNatRules/swazspfe00winrmin')]"
                },
                {
                  "id": "[concat(resourceId('Microsoft.Network/loadBalancers', 'lb0'),'/inboundNatRules/swazspfe00winrmhttpsin')]"
                },
                {
                  "id": "[concat(resourceId('Microsoft.Network/loadBalancers', 'lb0'),'/inboundNatRules/swazspfe00sshin')]"
                }
              ],
              "loadBalancerBackendAddressPools": [
                {
                  "id": "[concat(resourceId('Microsoft.Network/loadBalancers', 'lb0'), '/backendAddressPools/0lbbc')]"
                }
              ],
              "primary": true,
              "subnet": {
                "id": "[resourceId('Microsoft.Network/virtualNetworks/subnets', 'SharePointDscDev123', 'default')]"
              },
              "privateIPAllocationMethod": "Static",
              "privateIPAddressVersion": "IPv4",
              "privateIPAddress": "192.168.6.7"
            }
          }
        ],
        "enableIPForwarding": false
      },
      "type": "Microsoft.Network/networkInterfaces",
      "apiVersion": "2022-01-01",
      "location": "[resourceGroup().location]"
    },
    {
      "dependsOn": [
        "[resourceId('Microsoft.Compute/availabilitySets', 'SharePointDscDev123')]",
        "[resourceId('Microsoft.Network/networkInterfaces', 'swazspfe00nic0')]"
      ],
      "properties": {
        "storageProfile": {
          "osDisk": {
            "createOption": "FromImage",
            "osType": "Windows",
            "caching": "ReadWrite"
          },
          "imageReference": {
            "publisher": "MicrosoftWindowsServer",
            "version": "latest",
            "offer": "WindowsServer",
            "sku": "2012-R2-Datacenter"
          },
          "dataDisks": []
        },
        "networkProfile": {
          "networkInterfaces": [
            {
              "id": "[resourceId('Microsoft.Network/networkInterfaces', 'swazspfe00nic0')]",
              "properties": {
                "primary": true
              }
            }
          ]
        },
        "osProfile": {
          "allowExtensionOperations": true,
          "computerName": "swazspfe00",
          "adminUsername": "Install",
          "adminPassword": "Somepass1",
          "windowsConfiguration": {
            "enableAutomaticUpdates": true,
            "provisionVMAgent": true,
            "winRM": {
              "listeners": [
                {
                  "protocol": "Http"
                }
              ]
            }
          }
        },
        "hardwareProfile": {
          "vmSize": "Standard_D2s_v3"
        },
        "availabilitySet": {
          "id": "[resourceId('Microsoft.Compute/availabilitySets', 'SharePointDscDev123')]"
        }
      },
      "apiVersion": "2022-03-01",
      "tags": {
        "AutomatedLab": "SharePointDscDev123",
        "CreationTime": "2022-11-30 11:46:39"
      },
      "location": "[resourceGroup().location]",
      "type": "Microsoft.Compute/virtualMachines",
      "name": "swazspfe00"
    }
  ],
  "parameters": {}
}
```

## Steps to reproduce

```powershell
Add-LabMachineDefinition -Name $SPFE1Name -AzureProperties @{RoleSize="Standard_D2s_v3"; StorageSku = "Standard_LRS"} -OperatingSystem $SPFE1Os -PostInstallationActivity $fePostInstallActivity;
Install-Lab;
```

## Environment details

[powershell_support_pack_2022_11_30-11_48_35.zip](https://github.com/AutomatedLab/AutomatedLab/files/10122292/powershell_support_pack_2022_11_30-11_48_35.zip)

PowerShell Version
```output
Name                           Value
----                           -----
PSVersion                      5.1.19041.1682
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1682
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

AutomatedLab components
```output
Name                      Version
----                      -------
AutomatedLab              5.45.0
AutomatedLab.Common       2.2.247
AutomatedLabDefinition    5.45.0
AutomatedLabNotifications 5.45.0
AutomatedLabTest          5.45.0
AutomatedLabUnattended    5.45.0
AutomatedLabWorker        5.45.0
```

--- a/AutomatedLabWorker/AutomatedLabAzureWorkerVirtualMachines.psm1
+++ b/AutomatedLabWorker/AutomatedLabAzureWorkerVirtualMachines.psm1
@@ -466,7 +466,7 @@
 
         if (-not $Lab.AzureSettings.IsAzureStack)
         {
-            $loadbalancer.outboundRules = @(
+            $loadbalancer.properties.outboundRules = @(
                 @{
                     name       = "InternetAccess"
                     properties = @{
@@ -737,7 +737,22 @@
                         createOption = "FromImage"
                         osType       = "Windows"
                         caching      = "ReadWrite"
-                    }
+                        managedDisk = @{
+                            storageAccountType = if ($Machine.AzureProperties.ContainsKey('StorageSku') -and $Machine.AzureProperties['StorageSku'] -notmatch 'ultra')
+                            {
+                                $Machine.AzureProperties['StorageSku']
+                            }
+                            elseif ($Machine.AzureProperties.ContainsKey('StorageSku') -and $Machine.AzureProperties['StorageSku'] -match 'ultra')
+                            {
+                                Write-ScreenInfo -Type Warning -Message "Ultra_SSD SKU selected, defaulting to Premium_LRS for OS disk."
+                                'Premium_LRS'
+                            }
+                            else
+                            {
+                                'Standard_LRS'
+                            }
+                        }
+                    }                    
                     imageReference = $imageRef
                     dataDisks      = @()
                 }
@@ -867,7 +882,13 @@ function Get-LWAzureVmSize
 
     $lab = Get-Lab
 
-    if ($machine.AzureProperties.RoleSize)
+    if ($machine.AzureRoleSize)
+    {
+        $roleSize = $lab.AzureSettings.RoleSizes |
+        Where-Object { $_.Name -eq $machine.AzureRoleSize }
+        Write-PSFMessage -Message "Using specified role size of '$($roleSize.Name)'"
+    }
+    elseif ($machine.AzureProperties.RoleSize)
     {
         $roleSize = $lab.AzureSettings.RoleSizes |
         Where-Object { $_.Name -eq $machine.AzureProperties.RoleSize }

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,8 @@
 
 - Fixed using wrong variable in `Wait-LWLabJob` (#1425).
 - Fixed output of `Add-LabDomainDefinition` (#1428)
+- Fixed OS disk ignoring Storage SKU (#1434)
+- Fixed outbound NAT rules attached to wrong resource
 
 ## 5.46.0 (2022-11-24)
 

